src/Microsoft.TestPlatform.AdapterUtilities/ManagedNameUtilities/ManagedNameHelper.Reflection.cs

```diff
@@ -142,10 +140,13 @@ public static void GetManagedName(MethodBase method!!, out string managedTypeNam
 
         managedTypeName = typeBuilder.ToString();
         managedMethodName = methodBuilder.ToString();
-        hierarchyValues = new[] {
+        hierarchyValues = hierarchyPos != null
```

Comment:
It's not clear what we want to do when `hierarchyPos` is null so I went with returning null for `hierarchyValues` but maybe we could throw?

Reply:
No, that can never happen. It always has a value. `AppendTypeString` only returns null if provided type is null, here it's not.